carta/image.py

```diff
@@ -73,6 +75,14 @@ def new(cls, session, directory, file_name, hdu, append, image_arithmetic, make_
         """
         command = "appendFile" if append else "openFile"
 
+        path = session.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+
+        if complex and image_arithmetic:
+            file_name = f'{expression}("{file_name}")'
```

Comment:
You unnecessarily moved this part of the expression logic into `new`. This part should *not* be here; it should only be in the session function for opening a complex image. This is only applicable to complex images. If you are opening a LEL expression, the expression parameter here is something completely different (an arbitrary string expression, not a constant which is used to construct the expression).

What you were previously passing to this function (a directory and a file name, where the file name could also be a string expression) was correct. Please revert most of these changes. The *only* thing that you should be moving in here is the `resolve_file_path` step, which you should be calling *on the directory parameter* that you are passing in here.

`open_image` and `open_complex image` should take a single path parameter and split it into a directory and filename, and `open_LEL_image` should take a separate directory and expression parameter. This was correct before.